compliance/virtualmachines/relay/relay.go

```diff
@@ -49,21 +63,42 @@ func (s *VsockServer) Stop() {
 	}
 }
 
+func (s *VsockServer) acquireSemaphore(parentCtx context.Context) error {
+	semCtx, cancel := context.WithTimeout(parentCtx, s.maxSemaphoreWaitTime)
+	defer cancel()
+
+	metrics.VsockSemaphoreQueueSize.Inc()
+	defer metrics.VsockSemaphoreQueueSize.Dec()
+	if err := s.semaphore.Acquire(semCtx, 1); err != nil {
+		if errors.Is(err, context.DeadlineExceeded) {
+			metrics.VsockConnectionsRejected.Inc()
```

Comment:
shouldn't this be out of the `if errors.Is`?

Reply:
That sounds good, I took a shot at it here: https://github.com/stackrox/stackrox/pull/17243/commits/013998595d1810aed6236267cb62d56ce59cbb0c